controllers/azuremachine_reconciler.go

```diff
@@ -252,6 +252,13 @@ func (s *azureMachineService) reconcileVirtualMachine(ctx context.Context, nicNa
 			if err != nil {
 				return nil, errors.Wrapf(err, "failed to delete machine")
 			}
+			OSDiskSpec := &disks.Spec{
+				Name: azure.GenerateOSDiskName(s.machineScope.Name()),
+			}
+			err = s.disksSvc.Delete(ctx, OSDiskSpec)
```

Comment:
/hold

I probably need to do the same for VMSS

Reply:
I don't think you do. We don't specify the disk name with VMSS.

Reply:
that's true

/hold cancel